### PR TITLE
0.1.1

### PR DESCRIPTION
**Pull request by sigmafelix**
- MODIS calculation function + tests
- Base learners: XGBoost and random forests
- Documentation fix: no titles in index if multiple lines of title in roxygen2
- Tested previously uncovered lines
- Removed the startup message
- MODIS calculation function test data are RAW data, potentially replaced with subsets later

### CODE REVIEW

**Comment by sigmafelix** [comment]
Function profiling of modis_mosaic_mod06:
image

@Spatiotemporal-Exposures-and-Toxicology MOD06 processing function uses `stars` functions and test errors are from that function.`lwgeom` package is a suggested package from `stars`, which is called only if longitude-latitude CRS is used. Since it is a suggested package, it is not installed in the r-dependencies action. I will make another PR for modifying DESCRIPTION in `main` branch to figure this problem out.

**Comment by Spatiotemporal-Exposures-and-Toxicology** [comment]
@sigmafelix STARS is needed for the handling of curvilinear grids?

**Comment by sigmafelix** [comment]
@Spatiotemporal-Exposures-and-Toxicology Yes, there are two reasons. First, even though `terra::rectify()` can assign curvilinear grids into regular grids, it detects the extent incorrectly (as introduced in #197 ). `stars` properly handles the process of rectifying/warping curvilinear grids. Second, we have to wait `terra` updates on CRAN for GitHub Action, even if `terra::rectify` issue could be fixed soon. Therefore, I warped the curvilinear grid data with `stars::st_warp()` then converted it to SpatRaster to mosaic multiple files into one SpatRaster object. Mosaicking in stars returned a value error (also mentioned in #197).

**Comment by sigmafelix** [comment]
Additional dependency is found: `FNN` package. Continue to work on this later. `FNN` is also in Suggests: in `stars` package.
https://github.com/r-spatial/stars/blob/666abc7a92f0a587d7d16d43a8ba654f1c8efb63/R/warp.R#L74-L127

**Comment by sigmafelix** [comment]
This PR is on hold to improve coverage and I will try to make another push before Friday.

**Comment by sigmafelix** [comment]
Does `withr` error in GH actions needs adding specific lines of code or something @mitchellmanware ? I am seeing the same error in `withr` 3.0.0 issue in #264 
Thank you.

**Comment by mitchellmanware** [comment]
My pull request yesterday worked fine after `withr` issues resolved (
https://github.com/r-lib/withr/issues/253). If the PR-related breaks
continue with the package maybe bring it up again on the thread.

**Comment by sigmafelix** [comment]
@Spatiotemporal-Exposures-and-Toxicology All checks passed and I think it can be merged into main.
For records, I removed a cached status for Github Action (in Actions->Caches), which stored the r-dependencies before `withr` error was resolved. DESCRIPTION in the incoming branch seems to be worked okay with R CMD check as opposed to the assumption that GitHub Action works with the DESCRIPTION file in the `main` branch.

File: R/calculate_covariates_support.R
Comment (Spatiotemporal-Exposures-and-Toxicology) [inline R/calculate_covariates_support.R:1]: @sigmafelix I looked at what MOD11A1 is and there is a warning about versions being deprecated.

https://lpdaac.usgs.gov/products/mod11a1v006/

Are all of the products included here (MOD11A1, MOD13A2, ...) still currently active and expected to be available for 2023, 2024, ... ?

Comment (sigmafelix) [reply R/calculate_covariates_support.R:1]: @Spatiotemporal-Exposures-and-Toxicology We are currently using [version 6.1](https://lpdaac.usgs.gov/products/mod11a1v061/), which was updated last year. The last version (6) rolled out in 2016, so I think the next version may be coming a few years later. Actually, MODIS Terra/Aqua already [exceeded its expected lifetime by seven years](https://ntrs.nasa.gov/citations/20230009857). We might prepare for the alternative data source in the long run.

**Comment by sigmafelix** [comment]
@Spatiotemporal-Exposures-and-Toxicology I will merge two PRs by Mitchell and Ranadeep soon.

**Comment by Spatiotemporal-Exposures-and-Toxicology** [comment]
@sigmafelix Thanks!

**Comment by sigmafelix** [comment]
@Spatiotemporal-Exposures-and-Toxicology I see a pkgdown error:

> ℹ pkgdown can only use images in 'man/figures' and 'vignettes' 

This is what I did not see in the test results from previous pushes and no changes were made into the cross_validation_types.Rmd. Have you seen this error before?

**Comment by sigmafelix** [comment]
To deal with this problem, we would have to use the explicit file paths required by pkgdown (e.g., figure files in /vignettes or /man/figures). It seems that pkgdown does not like raw R markdown files with figures, but I am still confused with the previous test passes. I will work on pre-knitting vignettes with separate figure files stored in man/figures.

**Review by Spatiotemporal-Exposures-and-Toxicology — APPROVED** [review]

**Comment by sigmafelix** [comment]
For records: https://github.com/r-lib/pkgdown/issues/1943
It seems like `pkgdown` takes Rmd files in `./vignettes` and points its html outputs to the root directory. That is, the figure files need to be referred to from the package root directory (i.e., when using relative paths). However, the error was not resolved following the guidance. Perhaps _pkgdown.yml or other internal settings were messed up when I merged other folks' branches.

**Comment by Spatiotemporal-Exposures-and-Toxicology** [comment]
@sigmafelix the _pkgdown.yml in the root is a relative recent addition and overrides functions of the .github/workflows/pkgdown.yaml. If you create hotfix branch and test removing the _pkgdown, we could at least isolate and see if that file is the source.

**Comment by sigmafelix** [comment]
@Spatiotemporal-Exposures-and-Toxicology I realized that the error came from download_data.Rmd due to changes in R/download.R. A new finding is that the same warning message about figure paths also appeared in the previous successful PRs. 

I will do:
1. Rollback to the ordinary Rmd (not pre-knitted) of the cross validation vignette
2. Reflect changes in R/download.R on vignettes/download_data.Rmd

**Pull request merged**